crates/symbolicator-service/src/services/ppdb_caches.rs

```diff
@@ -142,26 +169,38 @@ impl PortablePdbCacheActor {
                 scope: request.scope.clone(),
                 purpose: ObjectPurpose::Debug,
             })
-            .await
-            .map_err(|e| Arc::new(PortablePdbCacheError::Fetching(e)))?;
+            .await else {
+            return (CacheEntry::InternalError, AllObjectCandidates::default(), ObjectFeatures::default())
+        };
 
         match meta {
             Some(handle) => {
-                self.ppdb_caches
+                match self
+                    .ppdb_caches
                     .compute_memoized(FetchPortablePdbCacheInternal {
                         request,
                         objects_actor: self.objects.clone(),
                         object_meta: handle,
-                        candidates,
+                        candidates: candidates.clone(),
                     })
                     .await
+                {
+                    Ok(ppdb_cache_file) => {
+                        let PortablePdbCacheFile {
+                            status,
+                            data,
+                            candidates,
+                            features,
+                            ..
+                        } = Arc::try_unwrap(ppdb_cache_file).unwrap_or_else(|arc| (*arc).clone());
+                        let cache_entry = CacheEntry::from((status, data));
+                        let mapped = cache_entry.try_map(parse_ppdb_cache_owned);
+                        (mapped, candidates, features)
+                    }
+                    Err(e) => (e.as_ref().into(), candidates, ObjectFeatures::default()),
```

Comment:
in theory the object meta handle should have `features`. Similar thing applies here as above: We don’t need to pass those along through the whole `FetchPortablePdbCacheInternal`.